blocks/footer/footer.js

```diff
@@ -39,122 +39,136 @@ export default async function decorate(block) {
   const footer = document.createElement('div');
 
   // Footer content - Store Switcher
-  footer.innerHTML = `
+  // Check if root metadata exists for multistore
+  const hasRootMeta = getMetadata('root');
+
+  if (hasRootMeta) {
+    footer.innerHTML = `
       <div class="storeview-switcher-button"></div>
     `;
 
-  // Container and component refs
-  let modal;
-
-  // Modal Actions
-  const showModal = async (content) => {
-    modal = await createModal([content]);
-    modal.showModal();
-  };
-
-  // Rendering the Store Switcher Modal Content
-  const $storeSwitcherBtn = footer.querySelector(
-    '.storeview-switcher-button',
-  );
-
-  // Store Switcher Modal Content - Set RootRoot Path to true
-  const storeSwitcherPath = '/store-switcher';
-  const fragmentStoreView = await loadFragment(storeSwitcherPath);
-  const storeSwitcher = document.createElement('div');
-
-  // Return Storename from stores-switcher
-  const selected = [...fragmentStoreView.querySelectorAll('a')].find((a) => {
-    const url = new URL(a.href);
-    return url.pathname.startsWith(root);
-  });
-
-  storeSwitcher.id = 'storeview-modal';
-  while (fragmentStoreView.firstElementChild) {
-    storeSwitcher.append(fragmentStoreView.firstElementChild);
-  }
+    // Container and component refs
+    let modal;
+
+    // Modal Actions
+    const showModal = async (content) => {
+      modal = await createModal([content]);
+      modal.showModal();
+    };
+
+    // Rendering the Store Switcher Modal Content
+    const $storeSwitcherBtn = footer.querySelector(
+      '.storeview-switcher-button',
+    );
+
+    // Store Switcher Modal Content
+    const storeSwitcherPath = '/store-switcher';
+    let fragmentStoreView;
+
+    try {
```

Comment:
@cctina516 try catch statement was made for error feedback when /storeswitcher is unavailable or unpublished